### DISCUSSION TITLE
How do you draw these points into a line in pole figure

### DISCUSSION THREAD

**Discussion by xiu-shen** [mtex-toolbox/mtex#2265] [Ask Anything]
![image](https://github.com/user-attachments/assets/db63edec-b32b-45b2-b251-d8a2081fb6f0)

**Comment by zmichels** [comment]
Do any of these help?

plot(h1, 'plane')

Or...

circle(h1)

**Reply by xiu-shen** [reply]
It doesn't seem to take into account which polar plane I'm drawing. I want to draw multiple projections of a certain orientation rotating along a particular axis such as ori*h1 projections on 0001 or 10-10 polar graphs

**Reply by xiu-shen** [reply]
> Do any of these help?
> 
> plot(h1, 'plane')
> 
> Or...
> 
> circle(h1)

Hello, bother you again, this problem has not been solved at present

**Reply by zmichels** [reply]
> > Do any of these help?
> 
> > 
> 
> > plot(h1, 'plane')
> 
> > 
> 
> > Or...
> 
> > 
> 
> > circle(h1)
> 
> 
> 
> Hello, bother you again, this problem has not been solved at present

Do you mind sharing the small dataset? It is not a difficult task, but perhaps it depends on specifics in your data to get something like you want.

**Reply by xiu-shen** [reply]
mtexdata twins
figure;
[~, mP] = plot(ebsd, ebsd.orientations);
mP.micronBar.visible = 'off';
hold on;
[grains, ebsd.grainId, ebsd.mis2mean] = calcGrains(ebsd('indexed'), 'angle', 10 * degree);
plot(grains.boundary, 'linewidth', 1);
hold off;

g = ginput(1);
grainID = grains.findByLocation(g);
ori = ebsd(ebsd.findByLocation(g)).orientations;

h = Miller(0, 0, 0, 1, cs);
h1 = Miller(0, 1, -1, 0, cs, 'uvw');
angles = -90:10:90; 
colors = lines(length(angles)); 
figure;
plotPDF(ori, h, 'DisplayName', 'Original', 'MarkerSize', 16, 'MarkerColor', 'g', 'grid', 'grid_res', 90 * degree, 'antipodal');
hold on;

for i = 1:length(angles)
    Tori_theory = ori * rotation.byAxisAngle(h1, angles(i) * degree);
    plotPDF(Tori_theory, h, 'DisplayName', ['Angle ' num2str(angles(i))], ...
            'MarkerSize', 16, 'MarkerColor', colors(i, :), 'grid', 'grid_res', 90 * degree, 'antipodal');
end
hold off;
This is the entire code

**Reply by xiu-shen** [reply]
> > > Do any of these help?
> > 
> > 
> > > 
> > 
> > 
> > > plot(h1, 'plane')
> > 
> > 
> > > 
> > 
> > 
> > > Or...
> > 
> > 
> > > 
> > 
> > 
> > > circle(h1)
> > 
> > 
> > Hello, bother you again, this problem has not been solved at present
> 
> Do you mind sharing the small dataset? It is not a difficult task, but perhaps it depends on specifics in your data to get something like you want.

I'm sorry to bother you again, but what should I do

**Comment by ralfHielscher** [comment]
```matlab
f = fibre(h1,h1)
plotPDF(f,Miller(0,0,0,1,cs))
```

**Reply by zmichels** [reply]
These approaches presume in your application that you have/know "h1" for plotting purposes. If instead, you want to fit a geodesic arc to a distribution/set of orientations, you could try something comprehensive like principal geodesic analysis (PGA) [PGA](https://github.com/zmichels/CVA/blob/master/PGA.m). [Fletcher et al., 2004](https://www.cs.jhu.edu/~misha/ReadingSeminar/Papers/Fletcher04.pdf)

**Reply by xiu-shen** [reply]
![image](https://github.com/user-attachments/assets/5dd89f2a-c380-4d3f-a211-cec832630d8e)
When drawing without considering grain orientation, only one triangle is drawn
When I consider the orientation of individual grains, matlab reports an error
Incorrect use of symmetry/ensureCS
The above two symmetries does not match!

Error fibre/plotPDF (line 44)
for i = 1:length(h), h{i} = f.CS.ensureCS(h{i});  end

**Reply by xiu-shen** [reply]
Hello, bother you again, this problem has not been solved at present

**Comment by zmichels** [comment]
```matlab
%% Import Data
mtexdata twins

% plot to select grain
figure;
[~, mP] = plot(ebsd, ebsd.orientations);
mP.micronBar.visible = 'off';
hold on;
[grains, ebsd.grainId, ebsd.mis2mean] = calcGrains(ebsd('indexed'), 'angle', 10 * degree);
plot(grains.boundary, 'linewidth', 1);
hold off;

% select grain
g = ginput(1);
grainID = grains.findByLocation(g);
ori = ebsd(ebsd.findByLocation(g)).orientations;

%% Miller info
cs = ori.CS;
h = Miller(0, 0, 0, 1, cs);
h1 = Miller(0, 1, -1, 0, cs, 'uvw');

%% angles for points to plot
angles = 0:10:180;

%% compute rotated orientations
Tori = rotation.byAxisAngle(ori*h1, angles * degree)*ori;

% colormap
colors = turbo(length(angles));

%% Make figure
figure;
% plot rotation-axis great-circle
plot(ori*h1, 'plane','LineColor','k','LineWidth',3,' antipodal','lower')
hold on

% loop through rotated orientations to plot with the colors
for i = 1:length(Tori)
    plot(Tori(i)*h,'antipodal','lower','MarkerFaceColor',colors(i,:),'MarkerEdgeColor','none','MarkerSize',16)
end
% add  original
plot(ori*h, 'DisplayName', 'Original', 'Marker', 's', 'MarkerSize', 20, 'MarkerFaceColor', 'w', 'MarkerEdgeColor','k','antipodal','lower');
% add rotation axis
plot(ori*h1,'DisplayName', 'Axis', 'Marker', '^', 'MarkerSize', 13, 'MarkerFaceColor', 'k', 'MarkerEdgeColor','k','antipodal','lower')
% add small circles projected from the rotation axis
circle(ori*h1,[10:10:80]*degree,' LineColor',[.5 .5 .5],'LineWidth',.5,'LineStyle','--','antipodal','lower')
hold off;
legend show
```

![connect_the_dots](https://github.com/user-attachments/assets/a40af1ab-9b53-425c-aa01-7994734a5ace)


Does something like this work to get you on a useful track?

**Reply by xiu-shen** [reply]
> ```matlab
> %% Import Data
> mtexdata twins
> 
> % plot to select grain
> figure;
> [~, mP] = plot(ebsd, ebsd.orientations);
> mP.micronBar.visible = 'off';
> hold on;
> [grains, ebsd.grainId, ebsd.mis2mean] = calcGrains(ebsd('indexed'), 'angle', 10 * degree);
> plot(grains.boundary, 'linewidth', 1);
> hold off;
> 
> % select grain
> g = ginput(1);
> grainID = grains.findByLocation(g);
> ori = ebsd(ebsd.findByLocation(g)).orientations;
> 
> %% Miller info
> cs = ori.CS;
> h = Miller(0, 0, 0, 1, cs);
> h1 = Miller(0, 1, -1, 0, cs, 'uvw');
> 
> %% angles for points to plot
> angles = 0:10:180;
> 
> %% compute rotated orientations
> Tori = rotation.byAxisAngle(ori*h1, angles * degree)*ori;
> 
> % colormap
> colors = turbo(length(angles));
> 
> %% Make figure
> figure;
> % plot rotation-axis great-circle
> plot(ori*h1, 'plane','LineColor','k','LineWidth',3,' antipodal','lower')
> hold on
> 
> % loop through rotated orientations to plot with the colors
> for i = 1:length(Tori)
>     plot(Tori(i)*h,'antipodal','lower','MarkerFaceColor',colors(i,:),'MarkerEdgeColor','none','MarkerSize',16)
> end
> % add  original
> plot(ori*h, 'DisplayName', 'Original', 'Marker', 's', 'MarkerSize', 20, 'MarkerFaceColor', 'w', 'MarkerEdgeColor','k','antipodal','lower');
> % add rotation axis
> plot(ori*h1,'DisplayName', 'Axis', 'Marker', '^', 'MarkerSize', 13, 'MarkerFaceColor', 'k', 'MarkerEdgeColor','k','antipodal','lower')
> % add small circles projected from the rotation axis
> circle(ori*h1,[10:10:80]*degree,' LineColor',[.5 .5 .5],'LineWidth',.5,'LineStyle','--','antipodal','lower')
> hold off;
> legend show
> ```
> 
> ![connect_the_dots](https://private-user-images.githubusercontent.com/8769619/399215301-a40af1ab-9b53-425c-aa01-7994734a5ace.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1Mzc5ODEsIm5iZiI6MTczNTUzNzY4MSwicGF0aCI6Ii84NzY5NjE5LzM5OTIxNTMwMS1hNDBhZjFhYi05YjUzLTQyNWMtYWEwMS03OTk0NzM0YTVhY2UucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMzBUMDU0ODAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2E0ZWUzNmNkODZlZDA0MDA3MGE2ZTc1MzhlODUyN2FlN2Q4ZGNkZGNmYjk2Njc1YWEyYjk3ZDRjNGM5YWJjMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.zo6LsVrfACN_LN0kgSK6E7HV71y-VfsmNnyud2_Hs_4)
> 
> Does something like this work to get you on a useful track?像这样的东西能让你走上有用的道路吗？

Thank you very much for your reply, but you ignored a problem that the lines drawn are consistent with the projection points in the pole diagram 0001. If I draw on the 11-20 diagram instead, then there should be 3 lines and how to change to the 11-20 pole diagram

**Reply by zmichels** [reply]
Sounds good. Give it a go!

**Reply by xiu-shen** [reply]
> Sounds good. Give it a go!

Sorry, I still don't understand how to draw three lines in a 10-12 pole diagram that are reoriented and rotate around the 01-10 axis

**Reply by zmichels** [reply]
What did you try that didn't work?

Here's something a little bit different that lets you specify a set of Miller axes. Maybe that's more like what you want -- its not entirely clear to me what you want or what isn't working.

```matlab
%% Import Data
mtexdata twins

% plot to select grain
figure;
[~, mP] = plot(ebsd, ebsd.orientations);
mP.micronBar.visible = 'off';
hold on;
[grains, ebsd.grainId, ebsd.mis2mean] = calcGrains(ebsd('indexed'), 'angle', 10 * degree);
plot(grains.boundary, 'linewidth', 1);
hold off;

% select grain
g = ginput(1);
grainID = grains.findByLocation(g);
ori = ebsd(ebsd.findByLocation(g)).orientations;

%% Miller info
cs = ori.CS;

% rotation axis
tAx = Miller(0, 1, -1, 0, cs, 'uvw');

% set all the Millers you might want
M = [Miller(0, 0, 0, 1, cs), Miller(1, 0, -1, 2, cs)];

%% angles for points to plot
angles = 0:10:180;

%% compute rotated orientations
Tori = rotation.byAxisAngle(ori*tAx, angles * degree)*ori;

% colormap
colors = turbo(length(angles));

%% Make figure

mfig = newMtexFigure;
% loop through different Miller axes
for k = 1: length(M)

    nextAxis(mfig, 1, k)
    % loop through rotated orientations to add plot of the symmetrised Millers
    for i = 1:length(Tori)
        plot(Tori(i)*symmetrise(M(k), cs),'antipodal','lower','MarkerFaceColor',colors(i,:),'MarkerEdgeColor','none','MarkerSize',16, 'figSize', 'medium')
        hold on
    end

    % add  original orientation's Miller(s)
    plot(ori*symmetrise(M(k), cs), 'DisplayName', 'Original', 'Marker', 's', 'MarkerSize', 20, 'MarkerFaceColor', 'w', 'MarkerEdgeColor','k','antipodal','lower');

    % add the rotation axis
    plot(ori*tAx,'DisplayName', 'Axis', 'Marker', '^', 'MarkerSize', 13, 'MarkerFaceColor', 'k', 'MarkerEdgeColor','k','antipodal','lower')

    % comput angles for lines to draw to connect millers
    angsFromAx = unique(round(angle(symmetrise(ori,cs)*M(k), ori*tAx, 'antipodal')./degree));

    % draw small circles projected at the calculated angles about the axis
    circle(ori*tAx,angsFromAx*degree,'antipodal','lower')
    hold off;
    mtexTitle(sprintf('%s', num2str(M(k).hkil)))
end
legend show
% end
% saveas(gcf, sprintf('connect_the_dots_%s.png',strrep(num2str(M2.hkil),' ','_')))
```
![connect_the_dots_1](https://github.com/user-attachments/assets/42faaae8-7ab6-4be1-bf2b-bc5efb9ce3af)
![connect_the_dots_2](https://github.com/user-attachments/assets/a03c48f6-fadc-4e45-ac06-03dedb5ed367)
![connect_the_dots_3](https://github.com/user-attachments/assets/54ce8213-b083-4111-a540-c718bb644101)
![connect_the_dots_4](https://github.com/user-attachments/assets/e38391d9-5d7a-45e3-8f70-edfb573f8501)

**Comment by zmichels** [comment]
Note:
Alternatively... if you had a set of orientations and weren't sure about what axis they might be rotationally related (i.e., you didn't construct the set from applying a specified rotation)... you could try something like principal geodesic analysis (PGA). Only note of caution is that recently some MTEX syntax changes in different versions of 5.x cause a need for an awkward  check in the code below. The PGA code section is from the PGA function at this repo:
[https://github.com/zmichels/CVA](url)

```matlab
ori = Tori;

%% PRINCIPAL GEODESIC ANALYSIS (PGA)
% PGA computes a dispersion tensor for a set of 3-or-more orientations
% the maximum eigenvector of the dispersion tensor provides a best-fit
% approximation for a rotation axis to describe the distribution of the
% orientation set.

% compute mean about which to project to tangent space
oriRef = mean(ori);

% compute the projection of orientations onto the three dimensional
% tangential space centered about the mean

% query MTEX version because silly syntax differences in different versions
ver = extract(getMTEXpref('version'),digitsPattern);

% adjust depending on version
if str2double(ver{2}) < 11
    t = log(ori, oriRef,'left');
else
    t = log(ori, oriRef,SO3TangentSpace.leftVector);
end

% covariance matrix as tensor
T = tensor(t*t,'rank',2);

% eigenvectors
[odt, mags] = eig(T);

% sort largest first for compatibility with previous versions
[~,ind] = sort (mags,'descend');

odt = odt(ind);
mags = mags(ind);

%% Make a plot
figure,
plot(Tori*h,'antipodal','lower','MarkerEdgeColor','none','MarkerSize',12)
hold on
% add best-fit rotation axis
plot(odt(1),'Marker','^','MarkerFaceColor','k', 'displayName','fit-rotation-axis')
% and the great circle
plot(odt(1),'plane', 'LineWidth',2)
% some projected small circles
circle(odt(1),[10:10:80]*degree, 'antipodal','lower', 'linestyle', '--','LineColor',[.5 .5 .5],'LineWidth',.5)
hold off
legend show
```

![PGA_connect_dots](https://github.com/user-attachments/assets/a002f836-6e2b-4681-b232-f375b8f2cdce)
